### DISCUSSION TITLE
Support multiple middlewares

### DISCUSSION THREAD

**Discussion by kyun** [vercel/next.js#45372] [Ideas]
### Describe the feature you'd like to request

I want to use middleware for multiple purposes like logging, authorization, etc.

But now only one `middleware.js` can exist.

### Describe the solution you'd like

Create multiple middlewares by routes or something else.

```
app 
  |-app.js
  |-layout.js
  |-template.js
  |-loading.js
  |-error.js
  |-middleware.js <-- This middleware is for root
  |-about
  |  |-app.js
  |  |- (...)
  |  |-middleware.js <-- This is for /about
```

### Describe alternatives you've considered

or create `/middlewares` directory and put all middleware into there.

```
middlewares
  |-withLogger.js
  |-withAuth.js
  |-index.js. <-- it is the root of middleware.
```

**withAuth.js**

```js
import { NextResponse } from 'next/server'

export function withLogger (request) {
    console.log('I am logging...');
    return NextResponse.next();
}

export const config = {
  matcher: ['/about', '/need_logging'],
}
```

**Comment by IonelLupu** [comment]
no updates here? did you find a nice solution to this issue @kyun ?

**Comment by asukhariev** [comment]
Hey guys, do you have any idea how to make work multiple middleware functions - especially next-auth and next-intl simultaneously ... Documentation is on ti's way long time already...
